src/client-python/reportclient/debuginfo.py

```diff
@@ -420,6 +418,10 @@ def download(self, files, download_exact_files=False):
 
             progress_observer.downloaded_pkgs += 1
 
+        for root, dirs, files in os.walk(self.cachedir):
+            for walked_dir in dirs:
+                os.chmod(os.path.join(root, walked_dir), 0o775)
```

Comment:
This will also not work for existing caches, right? Those will still be owned by abrt:abrt.

Reply:
Yes. That 775 is in fact just g+w. It does not alter the ownership. The ownership should be either abrt:abrt (from past) or UNPRIV:abrt (since this change). Both way it will work.

Reply:
I’m not sure it will.

If I have /var/cache/abrt-di already populated and owned by abrt:abrt with directories being 755:

1. I won’t be able to write there
2. I won’t be able to subsequently call `chmod`

There already is a thing in the abrt spec file that recursively chowns the cache as abrt:abrt.

Reply:
https://github.com/abrt/abrt/blob/434cfd609f8e9f76175d7c3160c764d837f9b73d/abrt.spec.in#L574 is what I’m talking about here.

Reply:
I also found my old code with this pearl:

```
+                        if os.path.islink(path):
+                            # fchmodat in glibc does not support AT_SYMLINK_NOFOLLOW
+                            # and it would otherwise fail because some links
+                            # under .build-id can be broken.
+                            continue
+                        stat_ = os.stat(path)
+                        os.chmod(path, stat_.st_mode | stat.S_IWGRP)
```

Though I can’t remember why I did that. I’ll try to see if it really fails with those links.

Reply:
Yeah, that `chown -R abrt:abrt` can be changed to `chown :abrt` - or even precisely, I think we can remove it as abrt 1 was looong ago. I wanted to do that in separate PR. Hmm, on the other hand I can replace it by `chmod -R 755`.

Reply:
Oh, right, you ignore regular files when changing permissions, so you can ignore that comment.